src/libAtomVM/externalterm.c

```diff
@@ -43,6 +44,7 @@
 #define SMALL_BIG_EXT 110
 #define EXPORT_EXT 113
 #define MAP_EXT 116
+#define ATOM_UTF8_EXT 118
```

Comment:
How did you make this kind of atoms?

Trying `erlang:term_to_binary('漢字')` gives me `<<131,119,6,230,188,162,229,173,151>>`.

Reply:
As you mentioned in chat, we can encode atoms with this encoding in the shell, e.g.,
```
4> rp(erlang:term_to_binary('∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y')).
<<131,118,1,168,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121,226,136,128,120,226,136,
  131,121,226,136,128,120,226,136,131,121,226,136,128,120,
  226,136,131,121,226,136,128,120,226,136,131,121,226,136,
  128,120,226,136,131,121,226,136,128,120,226,136,131,121,
  226,136,128,120,226,136,131,121>>
```

and also in the shell, we can decode back to an atom:

```
6> erlang:display(
6>         erlang:binary_to_term(
6>             <<131,118,1,168,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121,226,136,128,120,226,136,
6>             131,121,226,136,128,120,226,136,131,121,226,136,128,120,
6>             226,136,131,121,226,136,128,120,226,136,131,121,226,136,
6>             128,120,226,136,131,121,226,136,128,120,226,136,131,121,
6>             226,136,128,120,226,136,131,121>>
6>   )).
'∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y∀x∃y'
true
```

But we fail in AtomVM with

```
CRASH inary_to_term:
======
pid: <0.1.0>

Stacktrace:
[{test_binary_to_term,start,0,[{file,"/Users/fadushin/work/src/github/fadushin/AtomVM/tests/erlang_tests/test_binary_to_term.erl"},{line,27}]}]

cp: #CP<module: 0, label: 100, offset: 0>

x[0]: error
x[1]: badarg
x[2]: {1,1,90,1,[{0,23}],error}
```

(using a contrived example)

Reply:
ok, here is my point about this: if `ATOM_UTF8_EXT` is used **only** for encoding atoms that are **longer than 255 bytes**, there is no point in supporting it, since our atom table assumes that atoms have always length <= 255.

Conversely if `ATOM_UTF8_EXT` is used sometimes for encoding also short atoms (such as a 5 bytes atom), then it makes sense supporting it.

My assumption (that might be wrong) is that the most compact encoding is always used, so `SMALL_ATOM_UTF8_EXT` for the atoms that we can support.

Reply:
I agree, we can limit the implementation to not support `ATOM_UTF8_EXT`, as there is no appreciable use-case for it, especially for embedded systems.